ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/Runner.java

```diff
@@ -176,9 +176,7 @@ private RequestMapper createRequestMapper(
         new InternalResponseHandler<>(responseFactory, new EthSignResultProvider(signerProvider)));
     requestMapper.addHandler(
         "eth_signTransaction",
-        new InternalResponseHandler<>(
-            responseFactory,
-            new EthSignTransactionResultProvider(chainId, signerProvider, jsonDecoder)));
+        new SignTransactionHandler(chainId, signerProvider, responseFactory, transactionFactory));
```

Comment:
I think it might be clearer to create all the Handlers together (following sendTxHandler line 165) and condense the addHandler lines

Reply:
also why don't eth_accounts and eth_sign handlers need the chainId passed in?

Reply:
The transaction serializer requires it. Those methods don't do serializing.